### PR TITLE
refactor: handle arbitrary file types instead of hardcoding js and css

### PR DESCRIPTION
**Pull request by SimenB**
## JIRA Issue
[COREWEB-122](https://jira.finn.no/browse/COREWEB-122)

## Status
READY

## Description

This is a small refactor I did while implementing new endpoints. It can go in as a separate improvement to reduce the scope of the upcoming PR.

## Todos
- [x] Tests
- [ ] Documentation

## Related PRs
* None

### CODE REVIEW

File: test/main.test.js [hunk 1]
```diff
@@ -719,3 +722,19 @@ describe('bundling multiple css feeds', () => {
...
+    let post;
+
+    beforeEach(async () => {
+        const router = new Router();
+        ({ server } = await createTestServerFor(router.router()));
+
+        ({ post } = supertest(server)); // eslint-disable-line prefer-const
+    });
+
+    test('unknown filetype gives 404', () => post('/bundle/png').expect(404));
```
Comment (SimenB) [inline test/main.test.js]: 404 keeps the old behavior. Could be a 400, I suppose? IDK

File: lib/main.js [hunk 2]
```diff
@@ -33,10 +32,11 @@ class CheckEmptyPayload extends Transform {
...
             json = JSON.parse(buffered);
         } catch (err) {
-            const msg =
+            const message =
                 'Expected payload provided to `/feed` to be an array ' +
-                `of feed objects but instead payload was unparseable. Got ${buffered}`;
-            return cb(Boom.badRequest(msg));
+                `of feed objects but instead payload was unparseable. Got "${buffered}".`;
+
+            return cb(Boom.boomify(err, { statusCode: 400, message }));
```
Comment (SimenB) [inline lib/main.js]: This keeps the original error for our logging

File: test/main.test.js (hunk 1 above)
Comment (digitalsadhu) [reply test/main.test.js]: yeah agree could be either but unless you have strong feelings towards 400 I'd say lets just keep it the same and stick with 404?

File: lib/params.js [hunk 3]
```diff
@@ -27,3 +27,28 @@ module.exports.file = (req, res, next, param) => {
...
+ * Validate the "type" parameter
+ *
+ * @param {Object} req HTTP request object
+ * @param {Object} res HTTP response object
+ * @param {function} next Next function in the route
+ * @param {String} param Parameter from the URL
+ */
+
+module.exports.type = (req, res, next, param) => {
+    schemas.type.validate(param, (error, value) => {
```
Comment (digitalsadhu) [inline lib/params.js]: Any reason you are using the callback form of validate or just preference?

Comment (SimenB) [reply lib/params.js]: To keep with the style of the others.

File: lib/main.js (hunk 2 above)
Comment (SimenB) [reply lib/main.js]: I'll tag the commit as `fix` to get this improvement out

**Pull request merged**